### ISSUE TITLE
`Namespace.Interface.prop` gives wrong error message

### ISSUE THREAD

**Issue by Zzzen** [microsoft/TypeScript#45333]
# Bug Report

### 🔎 Search Terms

namespace interface property dot

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about _________

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ts=3.3.3#code/HYQwtgpgzgDiDGEAEA5c06KQbwFBIKQgA8YB7AJwBckBLYKiCgMwWQEkGnWs9D+kMCmRgAuJFCoV6Ac3yEAvriW4A9KqQBRCsIri0kWGyQByAxjYmkACxBQkwMkVKVGAEySQwAIyanOjCyWAHS4VACeMMgAQgCMSAC8qOhGiMEB3GzBQiIA3LhhkTEATInJhpgQ6VxBiADaJjkwJgC6+QW4JOTUdDU8HDjyBGJIIMDh+SrqWjqU4gDCY440CIhQ9ibs2Va+8CAArlDIm1a09iBIEVEANEje+zTLow4plcFIACK0HuFk+54QMaXJwUCBSWgQABuyCo1hhRSQZGYlzhgmEUWo4VMzV6-isAHdaLD-HUAEQwUktEwAfgAFIUooIylsYLkgA)

### 💻 Code

```ts
namespace Namespace {
    export interface Interface {
        prop: string
    }
}

// Error: Namespace 'Namespace' has no exported member 'Interface'.
type B1 = Namespace.Interface.prop;

type B2 = Namespace.Interface['prop'];



export interface I {
    p: any;
}

// Error: Cannot access 'I.p' because 'I' is a type, but not a namespace. Did you mean to retrieve the type of the property 'p' in 'I' with 'I["p"]'?(
type p = I.p;
```

### 🙁 Actual behavior

`type B1 = Namespace.Interface.prop;` gives `Namespace 'Namespace' has no exported member 'Interface'.`

### 🙂 Expected behavior

It should be `Cannot access 'Namespace.Interface.prop' because 'Namespace.Interface' is a type, but not a namespace. Did you mean to retrieve the type of the property 'prop' in 'Namespace.Interface' with 'Namespace.Interface["prop"]'?`

**Comment by DanielRosenwasser** [comment]
Thanks @Zzzen!
